sstables/sstables.cc

```diff
@@ -1396,6 +1398,32 @@ void sstable::write_filter() {
     write_simple<component_type::Filter>(filter_ref);
 }
 
+void sstable::update_total_reclaimable_memory() {
+    // Only the bloom filter memory is reclaimable for now
+    _total_reclaimable_memory = _components->filter ? _components->filter->memory_size() : 0;
+}
```

Comment:
Such `update` methods are always error prone. Let's make `_total_reclaimable_memory` `std::optional<>` instead, and update it in the getter if not set (mark it as `mutable` to keep the getter `const`).

Reply:
Ack. I have updated the code.